Orange/widgets/utils/signals.py

```diff
@@ -0,0 +1,242 @@
+import copy
+
+from Orange.canvas.registry.description import InputSignal, OutputSignal
+
+
+class _Signal:
+    @staticmethod
+    def get_flags(multiple, default, explicit, dynamic):
+        """Compute flags from arguments"""
+        from Orange.canvas.registry.description import \
+            Single, Multiple, Default, NonDefault, Explicit, Dynamic
+        return (Multiple if multiple else Single) | \
+                (Default if default else NonDefault) | \
+                (explicit and Explicit) | \
+                (dynamic and Dynamic)
+
+
+class Input(InputSignal, _Signal):
+    """
+    Description of an input signal.
+
+    The class is used to declare input signals for a widget as follows
+    (the example is taken from the widget Test & Score)::
+
+        class Inputs:
```

Comment:
Inputs and Outputs are not used as types, but as namespaces (We could drop the classes, put the signal defintions directly onto the widget and nothing would change - functionality wise). I do not consider them to be enums (links @kernc posted) but simply a way to group related attributes.

Plural looks better to me.

PS.: If this would not lead to even more discussion, I might even suggest using lowercase name (outputs, inputs), as the same name is used when it gets instantiated and one would expect lowercase name for instance variable.